Multi-match search missing bool_prefix type

As you can see with the new `search_as_you_type` field type, the recommended way to query such fields is with a Multi-match with a `bool_prefix` type (https://www.elastic.co/guide/en/elasticsearch/reference/current/search-as-you-type.html):

```
GET my_index/_search
{
  "query": {
    "multi_match": {
      "query": "brown f",
      "type": "bool_prefix",
      "fields": [
        "my_field",
        "my_field._2gram",
        "my_field._3gram"
      ]
    }
  }
}
```

Unfortunately `elastic-builder` complains if we try and use that though:

```
WildcardUtilsServer.ts:62 Error: The 'type' parameter should be one of Set(5) {
  'best_fields',
  'most_fields',
  'cross_fields',
  'phrase',
  'phrase_prefix'
}
    at C:\dev\_projects\vibescout-web-workspace\node_modules\elastic-builder\lib\core\util.js:163:15
```

So, this new multi-match type needs to be added in as an option in the library.

I'll try make a pull request now.

--- a/src/core/consts.js
+++ b/src/core/consts.js
@@ -52,7 +52,8 @@ exports.MULTI_MATCH_TYPE = new Set([
     'most_fields',
     'cross_fields',
     'phrase',
-    'phrase_prefix'
+    'phrase_prefix',
+    'bool_prefix'
 ]);
 
 exports.SCORE_MODE_SET = new Set([

--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -732,9 +732,11 @@ declare namespace esb {
          * the `_score` from each field.
          * - `phrase_prefix` - Runs a `match_phrase_prefix` query on each field
          * and combines the `_score` from each field.
+         * - `bool_prefix` - Creates a match_bool_prefix query on each field and
+         * combines the _score from each field.
          *
          * @param {string} type Can be one of `best_fields`, `most_fields`,
-         * `cross_fields`, `phrase`, and `phrase_prefix`. Default is `best_fields`.
+         * `cross_fields`, `phrase`, `phrase_prefix` and `bool_prefix`. Default is `best_fields`.
          */
         type(
             type:
@@ -743,6 +745,7 @@ declare namespace esb {
                 | 'cross_fields'
                 | 'phrase'
                 | 'phrase_prefix'
+                | 'bool_prefix'
         ): this;
 
         /**

--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -732,7 +732,7 @@ declare namespace esb {
          * the `_score` from each field.
          * - `phrase_prefix` - Runs a `match_phrase_prefix` query on each field
          * and combines the `_score` from each field.
-         * - `bool_prefix` - Creates a match_bool_prefix query on each field and
+         * - `bool_prefix` - (added in v7.2) Creates a match_bool_prefix query on each field and
          * combines the _score from each field.
          *
          * @param {string} type Can be one of `best_fields`, `most_fields`,

--- a/src/queries/full-text-queries/multi-match-query.js
+++ b/src/queries/full-text-queries/multi-match-query.js
@@ -114,6 +114,9 @@ class MultiMatchQuery extends FullTextQueryBase {
      * - `phrase_prefix` - Runs a `match_phrase_prefix` query on each field
      * and combines the `_score` from each field.
      *
+     * - `bool_prefix` - Creates a match_bool_prefix query on each field and
+     * combines the _score from each field.
+     *
      * @example
      * // Find the single best matching field
      * const qry = esb.multiMatchQuery(['subject', 'message'], 'brown fox')
@@ -141,7 +144,8 @@ class MultiMatchQuery extends FullTextQueryBase {
      *     .operator('and');
      *
      * @param {string} type Can be one of `best_fields`, `most_fields`,
-     * `cross_fields`, `phrase`, and `phrase_prefix`. Default is `best_fields`.
+     * `cross_fields`, `phrase`, `phrase_prefix` and `bool_prefix`. Default is
+     * `best_fields`.
      * @returns {MultiMatchQuery} returns `this` so that calls can be chained.
      */
     type(type) {

--- a/src/queries/full-text-queries/multi-match-query.js
+++ b/src/queries/full-text-queries/multi-match-query.js
@@ -114,7 +114,7 @@ class MultiMatchQuery extends FullTextQueryBase {
      * - `phrase_prefix` - Runs a `match_phrase_prefix` query on each field
      * and combines the `_score` from each field.
      *
-     * - `bool_prefix` - Creates a match_bool_prefix query on each field and
+     * - `bool_prefix` - (added in v7.2) Creates a match_bool_prefix query on each field and
      * combines the _score from each field.
      *
      * @example

